src/Lykke.Service.PayInternal.AzureRepositories/MerchantGroup/MerchantGroupRepository.cs

```diff
@@ -13,27 +14,34 @@ namespace Lykke.Service.PayInternal.AzureRepositories.MerchantGroup
     public class MerchantGroupRepository : IMerchantGroupRepository
     {
         private readonly INoSQLTableStorage<MerchantGroupEntity> _storage;
+        private readonly INoSQLTableStorage<AzureIndex> _groupIndexStorage;
         public MerchantGroupRepository(
-            INoSQLTableStorage<MerchantGroupEntity> storage)
+            INoSQLTableStorage<MerchantGroupEntity> storage,
+            INoSQLTableStorage<AzureIndex> groupIndexStorage
+            )
         {
             _storage = storage;
+            _groupIndexStorage = groupIndexStorage;
         }
-        public async Task<IMerchantGroup> GetAsync(string ownerMerchantId, string groupId)
+        public async Task<IMerchantGroup> GetAsync(string ownerId)
         {
-            return await _storage.GetDataAsync(MerchantGroupEntity.ByOwner.GeneratePartitionKey(ownerMerchantId), groupId);
+            AzureIndex index = await _groupIndexStorage.GetDataAsync(
+                MerchantGroupEntity.ByOwner.GeneratePartitionKey(ownerId),
```

Comment:
I believe this method was for group but not for owner as you can have multiple groups for one owner.